radis/io/hitran.py

```diff
@@ -328,6 +340,237 @@ def hit2df(
+
+    df = pd.DataFrame(LOCAL_TABLE_CACHE[os.path.basename(fname).split(".")[0]]["data"])
+    df.rename(
+        columns={
+            "molec_id": "id",
+            "local_iso_id": "iso",
+            "nu": "wav",
+            "sw": "int",
+            "a": "A",
+            "gamma_air": "airbrd",
+            "gamma_self": "selbrd",
+            "elower": "El",
+            "n_air": "Tdpair",
+            "delta_air": "Pshft",
+            "global_upper_quanta": "globu",
+            "global_lower_quanta": "globl",
+            "local_upper_quanta": "locu",
+            "local_lower_quanta": "locl",
+            "gp": "gp",
+            "gpp": "gpp",
+        },
+        inplace=True,
+    )
+
+    metadata = {}
+    # Last modification time of the original file :
+    metadata["last_modification"] = time.ctime(getmtime(fname))
+    if verbose >= 2:
+        print("Opening file {0} (cache={1})".format(fname, cache))
+        print("Last modification time: {0}".format(metadata["last_modification"]))
+    if load_wavenum_min and load_wavenum_max:
+        assert load_wavenum_min < load_wavenum_max
+
+    # Use cache file if possible
+    fcache = DataFileManager(engine).cache_file(fname)
+    if cache and exists(fcache):
+        relevant_if_metadata_above = (
+            {"wavenum_max": load_wavenum_min} if load_wavenum_min else {}
+        )  # not relevant if wavenum_max of file is < wavenum min required
+        relevant_if_metadata_below = (
+            {"wavenum_min": load_wavenum_max} if load_wavenum_max else {}
+        )  # not relevant if wavenum_min of file is > wavenum max required
+        from radis import __version__, config
+
+        df = load_h5_cache_file(
+            fcache,
+            cache,
+            valid_if_metadata_is=metadata,
+            relevant_if_metadata_above=relevant_if_metadata_above,
+            relevant_if_metadata_below=relevant_if_metadata_below,
+            current_version=__version__,
+            last_compatible_version=config["OLDEST_COMPATIBLE_VERSION"],
+            verbose=verbose,
+            engine=engine,
+        )
+        if df is not None:
+            return df
+    # Detect the molecule by reading the start of the file
+    try:
+        with open(fname) as f:
+            mol = get_molecule(int(f.read(2)))
+    except UnicodeDecodeError as err:
+        raise ValueError(
+            "You're trying to read a binary file {0} ".format(fname)
+            + "instead of an HITRAN file"
+        ) from err
+
+    # %% Post processing
+
+    # assert one molecule per database only. Else the groupbase data reading
+    # above doesnt make sense
+    nmol = len(df["id"].unique())
+    if nmol == 0:
+        raise ValueError("Databank looks empty")
+    elif nmol != 1:
+        # Crash, give explicity error messages
+        try:
+            secondline = df.iloc[1]
+        except IndexError:
+            secondline = ""
+        raise ValueError(
```

Comment:
Hello @Supriya1702 , I was wrong in asking to create a hit2df_hapi function.  The function does not convert a file to a DataFrame; since actually the file -> memory has been done by [fetch()  above](https://github.com/radis/radis/pull/505/files#diff-f1a2be23a785af89993778ad1bad43a66e9f8b0a2825be0060fef6434dc6b390R1411).  All the caching / metadata checks of hit2df are not required here. 

Therefore, hit2df_hapi should be removed., and your original implementation was better. 

Instead, the only thing you want to keep is the post-processing; . ie lines 475-532 (which includes the parsing). 
You can extract these lines in a new `post_process_hitran_data()` function, shared by `hit2df()`  in hitran.py; and by `download_all_hitran_isotopes` below

Reply:
@erwanp  I've done the changes, can you review them?